source/components/molecules/ImageUploader/ImageUploader.tsx

```diff
@@ -0,0 +1,278 @@
+      });
+  };
+  const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    setHorizontalScrollPercentage(
+      event.nativeEvent.contentOffset.x /
+        (event.nativeEvent.contentSize.width - event.nativeEvent.layoutMeasurement.width)
+    );
+  };
+
+  const validColorSchema = getValidColorSchema(colorSchema);
+  return (
+    <>
+      <Wrapper>
+        <Container horizontal onScroll={handleScroll} showsHorizontalScrollIndicator={false}>
+          {images.map((image, index) => (
+            <ImageItem
+              key={`${image.path}-${index}`}
+              filename={(image.path as string)}
+              onRemove={() => {
+                removeImageFromState(index);
+                deleteImageFromCloudStorage(index);
+              }}
+            />
+          ))}
+        </Container>
+        {images.length > 2 && <HorizontalScrollIndicator percentage={horizontalScrollPercentage} />}
+        <ButtonContainer>
+          <Button colorSchema={validColorSchema} onClick={toggleModal} disabled={maxImages && images.length >= maxImages}>
+            <Icon name="add" />
+            <Text> {buttonText && buttonText !== '' ? buttonText : 'Ladda upp bild'}</Text>
+          </Button>
+        </ButtonContainer>
+      </Wrapper>
+      <Modal visible={choiceModalVisible} hide={toggleModal} presentationStyle="overFullScreen" transparent animationType="fade">
+        <BackgroundBlur>
+          <PopupContainer colorSchema={validColorSchema} >
+            <Row>
+              <PopupLabel colorSchema={validColorSchema}>Lägg till bild</PopupLabel>
+              <TouchableOpacity onPress={toggleModal} activeOpacity={1}>
+                <Icon name="clear"/>
+              </TouchableOpacity>
+            </Row>
+            <PopupButton
+              colorSchema={validColorSchema}
+              block
+              variant="outlined"
+              onClick={() => { 
+                addImageFromCamera(); 
+                toggleModal();
+              }}
+            >
+              <Icon name="camera-alt" />
+              <Text>Kamera</Text>
+            </PopupButton>
+            <PopupButton
+              colorSchema={validColorSchema}
+              block
+              variant="outlined"
+              onClick={() => {
+                addImagesFromLibrary();
+                toggleModal();
+              }}
+            >
+              <Icon name="add-photo-alternate" />
+              <Text>Bildbibliotek</Text>
+            </PopupButton>
+          </PopupContainer>
+        </BackgroundBlur>
+      </Modal>
+    </>
+  );
+};
+
+ImageUploader.propTypes = {
+  /** The text on the upload button */
+  buttonText: PropTypes.string,
+  /** The images to initially populate the list with (i.e. their meta-data including uris) */
+  value: PropTypes.array,
+  /** What happens when either deleting or uploading an image, updating the form state */
+  onChange: PropTypes.func,
+};
```

Comment:
hmm

Reply:
I guess your not a fan of the comments?

Reply:
Fixed, but kept a comment about value, because I feel that it's not obvious what it is. (And before you complain about the name, calling it value is a convention across all our input components, used for dynamic rendering , so it's not something we can change)